addons/account/models/account_journal.py

```diff
@@ -722,6 +722,7 @@ def _get_journal_outstanding_payments_account_balance(self, domain=None, date=No
             ('display_type', 'not in', ('line_section', 'line_note')),
             ('move_id.state', '!=', 'cancel'),
             ('reconciled', '=', False),
+            ('journal_id', '=', self.id),
```

Comment:
This contradicts the docstring: 
"/!\ The current journal is not part of the applied domain. This is the expected behavior since we only want
        a logic based on accounts."

Is it really a bug?

Reply:
Discussed with AVW: he agreed we need now to add the journal into the filter since we allow to share the outstanding accounts accross multiple journals. I'll adapt the docstring